### DISCUSSION TITLE
When multiple extensions are enabled to provide inline suggestions how to show the name of the extension in hover UI when ghost text is displayed in editor

### DISCUSSION THREAD

**Discussion by ganeshrn** [microsoft/vscode-discussions#1056] [Extension Development QnA] [answered]
If I understand correctly VS Code merges the list of inline completions from all the extension that support it and activated for the file being edited. Is there a clean way to show the users from which extension the inline suggestion comes from when users scroll through the list of inline suggestions from hover UI using next/previous buttons. 

https://github.com/microsoft/vscode-discussions/assets/5292461/9e418053-0b95-4dc3-849f-be9d2be86697

In the above screen recording the first suggestion comes from copilot and the second suggestion is coming from Ansible Lightspeed extension. Is there a good way for users to know from which extension the inline suggestion is coming from.

For the inline suggestion displayed in editor is to possible to display the name of the extension in the hover UI?

**Comment by alefragnani** [comment, marked as answer]
Hi @ganeshrn ,

Since you can't define a _name/title_ for your provider, I suppose there is no way, today, to identify the provider that suggested each entry. I did a quick search in the VS Code extension's repo and didn't find any open issue that reflects your request, so I would suggest you to open a new one https://github.com/microsoft/vscode/issues/new?assignees=&labels=&projects=&template=feature_request.md.

IMHO, I think this new feature could be a good addition not only for the `InlineCompletionItemProvider`, but others as well.

Hope this helps

**Reply by ganeshrn** [reply]
Thanks @alefragnani for your response. I have raised an here https://github.com/microsoft/vscode/issues/205555

**Comment by theReynald** [comment]
Closing since conversation has come to an end.
